array_test.go

```diff
@@ -564,6 +566,56 @@ func TestArrayContainsOnly(t *testing.T) {
 	})
 }
 
+func TestArrayContainsAny(t *testing.T) {
+	reporter := newMockReporter(t)
+
+	value := NewArray(reporter, []interface{}{123, "foo"})
+
+	value.ContainsAny(123)
+	value.chain.assertOK(t)
+	value.chain.reset()
+
+	value.NotContainsAny(123)
+	value.chain.assertFailed(t)
+	value.chain.reset()
+
+	value.ContainsAny("foo", 123)
+	value.chain.assertOK(t)
+	value.chain.reset()
+
+	value.NotContainsAny("foo", 123)
+	value.chain.assertFailed(t)
+	value.chain.reset()
+
+	value.ContainsAny("foo", "foo")
+	value.chain.assertOK(t)
+	value.chain.reset()
+
+	value.NotContainsAny("foo", "foo")
+	value.chain.assertFailed(t)
+	value.chain.reset()
+
+	value.ContainsAny(123, "foo", "FOO")
+	value.chain.assertOK(t)
+	value.chain.reset()
+
+	value.NotContainsAny(123, "foo", "FOO")
+	value.chain.assertFailed(t)
+	value.chain.reset()
+
+	value.NotContainsAny("FOO")
+	value.chain.assertOK(t)
+	value.chain.reset()
+
+	value.ContainsAny([]interface{}{123, "foo"})
+	value.chain.assertFailed(t)
+	value.chain.reset()
```

Comment:
Seems this is the only check in TestArrayContainsAny where `ContainsOnly` fails. This check is good, but it's for "unusual" case (when the argument is array).

I suggest to add a check for "usual" case as well, like this:

```
value.ContainsAny("bar")
// assertFailed
```